src/base/bittorrent/session.cpp

```diff
@@ -1415,85 +1419,52 @@ void Session::configure(libtorrent::settings_pack &settingsPack)
-                       , 1 << libt::session::local_peer_class_id);
-            // loopback
-            f.add_rule(libt::address_v6::from_string("::1")
-                       , libt::address_v6::from_string("::1")
-                       , 1 << libt::session::local_peer_class_id);
-        }
-        catch (std::exception &) {}
-#endif
-    }
-    m_nativeSession->set_peer_class_filter(f);
+    // Assuming that libtorrent will not give 0 as class ID
+    if (m_UTPClassID != 0) return;
 
+    m_UTPClassID = m_nativeSession->create_peer_class("qbt utp class");
     libt::peer_class_type_filter peerClassTypeFilter;
     peerClassTypeFilter.add(libt::peer_class_type_filter::tcp_socket, libt::session::tcp_peer_class_id);
     peerClassTypeFilter.add(libt::peer_class_type_filter::ssl_tcp_socket, libt::session::tcp_peer_class_id);
     peerClassTypeFilter.add(libt::peer_class_type_filter::i2p_socket, libt::session::tcp_peer_class_id);
-    if (isUTPRateLimited()) {
-        peerClassTypeFilter.add(libt::peer_class_type_filter::utp_socket
-            , libt::session::local_peer_class_id);
-        peerClassTypeFilter.add(libt::peer_class_type_filter::utp_socket
-            , libt::session::global_peer_class_id);
-        peerClassTypeFilter.add(libt::peer_class_type_filter::ssl_utp_socket
-            , libt::session::local_peer_class_id);
-        peerClassTypeFilter.add(libt::peer_class_type_filter::ssl_utp_socket
-            , libt::session::global_peer_class_id);
-    }
+    peerClassTypeFilter.add(libt::peer_class_type_filter::utp_socket, m_UTPClassID);
+    peerClassTypeFilter.add(libt::peer_class_type_filter::ssl_utp_socket, m_UTPClassID);
+
+    peerClassTypeFilter.disallow(libt::peer_class_type_filter::utp_socket, libt::session::global_peer_class_id);
+    peerClassTypeFilter.disallow(libt::peer_class_type_filter::utp_socket, libt::session::local_peer_class_id);
+    peerClassTypeFilter.disallow(libt::peer_class_type_filter::ssl_utp_socket, libt::session::global_peer_class_id);
+    peerClassTypeFilter.disallow(libt::peer_class_type_filter::ssl_utp_socket, libt::session::local_peer_class_id);
+
     m_nativeSession->set_peer_class_type_filter(peerClassTypeFilter);
 }
 
+void Session::configureLimitsOnLAN()
+{
+    const bool unlimitedLAN = ignoreLimitsOnLAN();
+    const bool altSpeedLimitEnabled = isAltGlobalSpeedLimitEnabled();
+
+    libt::peer_class_info info = m_nativeSession->get_peer_class(libt::session_handle::local_peer_class_id);
+    info.download_limit = unlimitedLAN ? 0 : (altSpeedLimitEnabled ? altGlobalDownloadSpeedLimit() : globalDownloadSpeedLimit());
+    info.upload_limit = unlimitedLAN ? 0 : (altSpeedLimitEnabled ? altGlobalUploadSpeedLimit() : globalUploadSpeedLimit());
+    m_nativeSession->set_peer_class(libt::session_handle::local_peer_class_id, info);
+}
+
+void Session::configureLimitsOnUTP()
+{
+    // Assuming that libtorrent will not give 0 as class ID
+    if (m_UTPClassID == 0) return;
```

Comment:
Why this check? We should have correct id here. Please don't hide possible logical errors! Use Q_ASSERT instead.